streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignorTest.java

```diff
@@ -798,17 +798,43 @@ public void shouldReturnEmptyClusterMetadataIfItHasntBeenBuilt() throws Exceptio
     }
 
     @Test
-    public void shouldNotLoopInfinitelyOnMissingMetadataAndShouldNotCreateRelatedTasks() {
+    public void shouldCreateAllTasksAndInternalTopicIfAllInputTopicsAreKnown() {
+        runMetadataTest("topic1", true, "topic3", true);
+    }
+
+    @Test
+    public void shouldNotLoopInfinitelyOnPartlyMissingMetadataAndShouldNotCreateRelatedTasks1() {
```

Comment:
It just parametrization of the test... the overall setup is shared code... Not sure is `@parameter` would make it simpler?